File: Central Limit Theorem+Regression/python/solution_hw23.py
from utils import *;
####
# Path variables to files
#
####
path_to_data_folder = '../data' # DON'T CHANGE THIS PATH

#Q1
CoinFlip()


#Q2
# populationDistribution: string('uniform','normal')
# sampleSize: integer (~30)
# numberOfSamples: integer (>100)
populationDistribution = "normal";
sampleSize = 30;
numberOfSamples = 200;
CLT(populationDistribution,sampleSize,numberOfSamples)

#Q3a
SLR(path_to_data_folder+'/hw23R-Advertising.csv');

#Q3b
MLR(path_to_data_folder+'/hw23R-Advertising.csv')

#Q4
LogisticRegression(path_to_data_folder+'/hw23R-q4data.txt')

#Q5
LogisticRegressionImproved(path_to_data_folder+'/hw23R-q4data.txt')

File: Central Limit Theorem+Regression/R/solution_hw23.R
#cleanup before start
rm(list=ls(all=T))


####
# Path variables to files
#
####
setwd('.') #Inside code folder - DON'T CHANGE THIS PATH
#source('installPackages.R')
source('utils.R')
path_to_data_folder = '../data' # DON'T CHANGE THIS PATH 
library(MASS)
library(stats)

#Q1 
CoinFlip() 


#Q2
populationDistribution = 'normal'
sampleSize = 30
numberOfSamples = 120
CLT(populationDistribution,sampleSize,numberOfSamples)

#Q3a
SLR(paste(path_to_data_folder,'/hw23R-Advertising.csv',sep=''))

#Q3b 
MLR(paste(path_to_data_folder,'/hw23R-Advertising.csv',sep=''))

#Q4
LogisticRegression(paste(path_to_data_folder,'/hw23R-q4.txt',sep=''))

#Q5
LogisticRegressionImproved(paste(path_to_data_folder,'/hw23R-q4.txt',sep=''))

